guava/src/com/google/common/base/Preconditions.java

```diff
@@ -875,7 +903,9 @@ public static void checkState(
    * @see Verify#verifyNotNull Verify.verifyNotNull()
    */
   @CanIgnoreReturnValue
-  public static <T> T checkNotNull(T reference) {
+  @EnsuresNonNull("#1")
+  @Pure
+  public static <T extends @Nullable Object> @NonNull T checkNotNull(T reference) {
```

Comment:
checkNotNull should never be called on a null value.  You can remove all the annotations on this line, leaving it as it was before your changes.  The two added declaration annotations are good, though.